PowerEditor/src/ScitillaComponent/ScintillaEditView.cpp

```diff
@@ -2414,9 +2414,9 @@ void ScintillaEditView::convertSelectedTextTo(bool Case)
 		for (int i = 0 ; i < nbChar ; ++i)
 		{
 			if (Case == UPPERCASE)
-				selectedStrW[i] = (WCHAR)::CharUpperW((LPWSTR)selectedStrW[i]);
+				selectedStrW[i] = (WCHAR)(UINT_PTR)::CharUpperW((LPWSTR)selectedStrW[i]);
 			else
-				selectedStrW[i] = (WCHAR)::CharLowerW((LPWSTR)selectedStrW[i]);
+				selectedStrW[i] = (WCHAR)(UINT_PTR)::CharLowerW((LPWSTR)selectedStrW[i]);

```

Comment:
(sorry for the three separate posts!)

Some sort of wrapper function is definitely a good idea, albeit for a different PR. 

```
  return (wchar_t)(UINT_PTR)CharUpper((LPWSTR)c);
```
...however, still makes my eyes bleed.

Something roughly like:
```
return reinterpret_cast<wchar_t>(reinterpret_cast<UINT_PTR>(CharUpper(reinterpret_cast<LPWSTR>(c))));
```
....I think that's just a tiny bit better, because my head doesn't explode (as often) while parsing the expression with C's operator precedence rules.

> [...] my head doesn't explode (as often) [...]

Yeah, my head is just *that* volatile. As you might've guessed, I use a *LOT* of parenthesis.